src/client/Microsoft.Identity.Client/Utils/DateTimeHelpers.cs

```diff
@@ -63,15 +63,39 @@ public static long GetDurationFromWindowsTimestamp(string windowsTimestampInFutu
             return (long)unixTimestamp - CurrDateTimeInUnixTimestamp();
         }
 
-        public static long GetDurationFromNowInSeconds(string unixTimestampInFuture)
+        public static long GetDurationFromNowInSeconds(string dateTimeStamp)
         {
-            if (string.IsNullOrEmpty(unixTimestampInFuture))
+            if (string.IsNullOrEmpty(dateTimeStamp))
             {
                 return 0;
             }
 
-            long expiresOnUnixTimestamp = long.Parse(unixTimestampInFuture, CultureInfo.InvariantCulture);
-            return expiresOnUnixTimestamp - CurrDateTimeInUnixTimestamp();
+            // First, try to parse as Unix timestamp (number of seconds since epoch)
+            if (long.TryParse(dateTimeStamp, out long expiresOnUnixTimestamp))
+            {
+                return expiresOnUnixTimestamp - DateTimeHelpers.CurrDateTimeInUnixTimestamp();
+            }
+
+            // Try parsing as ISO 8601 
+            if (DateTimeOffset.TryParse(dateTimeStamp, null, DateTimeStyles.RoundtripKind, out DateTimeOffset expiresOnDateTime))
+            {
+                return (long)(expiresOnDateTime - DateTimeOffset.UtcNow).TotalSeconds;
+            }
+
+            // Try RFC 1123 format 
```

Comment:
Do we expect this? If not, pls don't add it.

Reply:
removed it